src/main/java/picard/sam/SamErrorMetric/CollectSamErrorMetrics.java

```diff
@@ -213,73 +262,99 @@ protected boolean requiresReference() {
         }
     }
 
-    private static final int MAX_DIRECTIVES = ReadBaseStratification.Stratifier.values().length + 1;
-    private static final Log log = Log.getInstance(CollectSamErrorMetrics.class);
-
-    @Override
-    protected int doWork() {
-
-        final Random random = new Random(42);
-
-        final ProgressLogger progressLogger = new ProgressLogger(log, 100000);
-        long nTotalLoci = 0;
-        long nSkippedLoci = 0;
-        long nProcessedLoci = 0;
-
-        IOUtil.assertFileIsReadable(INPUT);
-        IOUtil.assertFilesAreReadable(INTERVALS);
+    /**
+     * Process our input data by iterating through input variants and reads.
+     * Useful for going through whole genomes / exomes, but slow for sets of reads not beginning at the start of chr1.
+     *
+     * @return {@code 0} if all inputs were processed successfully.  {@code 1} if there was any kind of error.
+     */
+    private int processDataNormalIteration() {
 
-        final Collection<BaseErrorAggregation> aggregatorList = getAggregatorList();
         // Open up the input resources:
         try (
-                final SamReader sam = SamReaderFactory.makeDefault().referenceSequence(REFERENCE_SEQUENCE).open(INPUT);
-                final ReferenceSequenceFileWalker referenceSequenceFileWalker = new ReferenceSequenceFileWalker(REFERENCE_SEQUENCE);
-                final PeekableIterator<VariantContext> vcfIterator = new PeekableIterator<>(
-                        VCF == null ? Collections.emptyIterator() : new VCFFileReader(VCF, true).iterator())
+            final SamReader sam = SamReaderFactory.makeDefault()
+                    .referenceSequence(REFERENCE_SEQUENCE)
+                    .setOption(SamReaderFactory.Option.INCLUDE_SOURCE_IN_RECORDS, true)
```

Comment:
why are you putting the file source into the reads? you don't seems to be using this anywhere...

Reply:
Fixed.